external-crates/move/crates/move-vm-runtime/src/loader/ast.rs

```diff
@@ -33,6 +32,39 @@ use crate::{
 // Types
 //**************************************************************************************************
 
+pub type PackageStorageId = AccountAddress;
+pub type RuntimePackageId = AccountAddress;
+
+/// runtime_address::module_name::function_name
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct VTableKey {
+    pub package_key: RuntimePackageId,
+    pub module_name: Identifier,
+    pub function_name: Identifier,
+}
+
+/// A `PackageVTable` is a collection of function pointers indexed by the module and function name
+/// within the package.
+pub type PackageVTable = BinaryCache<(Identifier, Identifier), ArenaPointer<Function>>;
+
+/// Representation of a loaded package.
+pub struct LoadedPackage {
+    pub storage_id: PackageStorageId,
+    pub runtime_id: RuntimePackageId,
+
+    // NB: this is under the package's context so we don't need to further resolve by
+    // address in this table.
+    pub loaded_modules: BinaryCache<Identifier, LoadedModule>,
+
+    // NB: this is needed for the bytecode verifier. If we update the bytecode verifier we should
+    // be able to remove this.
+    pub compiled_modules: BinaryCache<Identifier, CompiledModule>,
```

Comment:
Can you explain what this. comment means?

Reply:
When we go to check linking we need to call into the bytecode verifier for checking compatibility across dependencies. For this it needs/uses the `CompiledModule` so we need to keep the `CompiledModule`s for the package around for this purpose until/if we change the cyclic check and linking checker in the bytecode verifier to work over loaded modules.